include/oneapi/dpl/type_traits

```diff
@@ -215,7 +216,10 @@ using ::std::is_volatile_v;
 using ::std::negation;
 using ::std::rank_v;
 using ::std::void_t;
-#endif
+#    if __cplusplus <= 201703L
+using ::std::is_literal_type_v;
```

Comment:
It is under two guards: `__cplusplus > 201402L` (c++17 and newer) and `__cplusplus <= 201703L` (c++17 and older), therefore it is used with c++17 only. I should probably change `__cplusplus <= 201703L` to ` __cplusplus == 201703L` to avoid confusion.